.travis.yml

```diff
@@ -551,26 +623,27 @@ matrix:
         - ulimit -n 8192
       script:
         - ./build-support/bin/travis-ci.sh -be
+
     # Rust on macOS
-    - name: "Rust + Platform-specific Tests OSX"
-      os: osx
+    - <<: *default_osx_test_config
+      name: "OSX Rust + platform-specific tests"
```

Comment:
We should split this shard back into two; one for rust tests (ci.sh -e) which uses the rust cache config, and possibly shares its cache with the bootstrap shard, and one which runs platform-specific tests (ci.sh -z) and uses the run cache config.

The shards were merged because the cost of bootstrapping rust was high, but now that that's ameliorated, we should split them so that they can benefit from more appropriate caching.

Reply:
Makes sense, but I'd rather do that in a separate change, since this one is already hefty. 

Also, as above, we can't share Travis caches, but we can maybe share the built rust state via S3? I don't know enough about what that state is in order to do so correctly, but either way, let's address in a followup?